edb/schema/referencing.py

```diff
@@ -546,8 +640,9 @@ def _build_alter_cmd_stack(self, schema, context, scls, *, referrer=None):
             else:
                 obj = None
 
-        cmd = delta
+        cmd: Union[sd.DeltaRoot, sd.AlterObject[Any]] = delta
         for obj in reversed(object_stack):
+            assert obj is not None
```

Comment:
Nitpick: `assert` after `get_referrer` instead would be cleaner.

Reply:
It looks like mypy is not checking so deeply. If I remove this assertion and move it after `get_referrer`, after the various operations; mypy complains that obj in the new iteration could be None.

In the screenshot below, note that the assertion is commented (it is placed after the call to get_referrer)

![image](https://user-images.githubusercontent.com/2576032/74675296-87f1fb80-51b3-11ea-9fca-bd62d4fe4bb5.png)

Reply:
OK